source/macos/AdaptiveCards/AdaptiveCards/Utils/ACSCollectionViewAlignLayout.swift

```diff
@@ -0,0 +1,272 @@
+        } else {
+            return self.itemsHorizontalAlignment
+        }
+    }
+    
+    func itemsVerticalAlignmentForSectionAtIndex(at section: Int?) -> ACSCollectionViewItemsVerticalAlignment {
+        if let section = section, let collectionView = self.collectionView, let delegate = collectionView.delegate as? ACSCollectionViewAlignLayoutDelegate {
+            return delegate.collectionView(collectionView, layout: self, itemsVerticalAlignmentInSection: section)
+        } else {
+            return self.itemsVerticalAlignment
+        }
+    }
+    
+    func itemsDirectionForSectionAtIndex(at section: Int?) -> ACSCollectionViewItemsDirection {
+        if let section = section, let collectionView = self.collectionView, let delegate = collectionView.delegate as? ACSCollectionViewAlignLayoutDelegate {
+            return delegate.collectionView(collectionView, layout: self, itemsDirectionInSection: section)
+        } else {
+            return self.itemsDirection
+        }
+    }
+}
+extension ACSCollectionViewAlignLayout {
+    func isCollectionViewNewLineStart(at indexPath: IndexPath) -> Bool {
+        if indexPath.item == 0 {
+            return true
+        }
+        guard indexPath.item > 0 else {
+            return false
+        }
+        let currentIndexPath = indexPath
+        let previousIndexPath = IndexPath(item: indexPath.item - 1, section: indexPath.section)
+        
+        let currentAttributes = super.layoutAttributesForItem(at: currentIndexPath)
+        let previousAttributes = super.layoutAttributesForItem(at: previousIndexPath)
+        let currentFrame = currentAttributes?.frame ?? .zero
+        let previousFrame = previousAttributes?.frame ?? .zero
+        
+        let insets = insetForSectionAtIndex(at: currentIndexPath.section)
+        let currentLineFrame = CGRect(x: insets.left, y: currentFrame.origin.y, width: self.collectionView?.frame.width ?? .zero, height: currentFrame.size.height)
+        let previousLineFrame = CGRect(x: insets.left, y: previousFrame.origin.y, width: self.collectionView?.frame.width ?? .zero, height: previousFrame.size.height)
+        
+        return !currentLineFrame.intersects(previousLineFrame)
+    }
+    
+    func lineAttributesArray(withStart startAttributes: ACSCollectionViewLayoutAttributes?) -> [ACSCollectionViewLayoutAttributes]? {
+        var lineAttributesArray: [ACSCollectionViewLayoutAttributes] = []
+        if let startAttributes = startAttributes {
+            lineAttributesArray.append(startAttributes)
+            guard let collectionView = self.collectionView else {
+                return nil
+            }
+            let itemCount = collectionView.numberOfItems(inSection: startAttributes.indexPath?.section ?? 0)
+            let insets = insetForSectionAtIndex(at: startAttributes.indexPath?.section)
+            var index = startAttributes.indexPath?.item ?? 0
+            
+            var isLineEnd = index == itemCount - 1
```

Comment:
```suggestion
            var isLineEnd = false
```
You don't need to add check for last element inside and outside loop